src/coreComponents/physicsSolvers/wavePropagation/AcousticFirstOrderWaveEquationSEM.cpp

```diff
@@ -337,13 +339,41 @@ void AcousticFirstOrderWaveEquationSEM::initializePostInitialConditionsPreSubGro
                                                                freeSurfaceFaceIndicator,
                                                                velocity,
                                                                damping );
+        //This portion of code work asd follow: compute the time-step then exit the code to let you put it inside the XML
+        if( m_preComputeDt==1 )
+        {
+          acousticFirstOrderWaveEquationSEMKernels::ComputeTimeStep< FE_TYPE > kernelT( finiteElement );
+
+          dtCompute = kernelT.template launch< EXEC_POLICY, ATOMIC_POLICY >( elementSubRegion.size(),
+                                                                             nodeManager.size(),
+                                                                             X,
+                                                                             elemsToNodes,
+                                                                             density,
+                                                                             mass );
+
+
+          real64 globaldt = MpiWrapper::min( dtCompute );
```

Comment:
Why are you taking the Min over all MPI ranks ? The deltaT (and the CFL condition) should be global, obtained by the Rayleigh product over the global fields, there shouldn't be a "local" timestep per rank, no?